Added a few things

KjellConnelly opened the pull request:
Added:
- mobile toolbar icons for commands
- ignore case for queries (helps on mobile which auto capitalizes)
- allows for date format picker with a few options: `YYYY-MM-DD`, `MM-DD-YYYY`, and `DD-MM-YYYY`.
- Added two new options to SettingsPlugin: `toggle` for allowing format changes (this is just for safety), and `dropdown` to pick one of the three.
- Updated README to explain consequences of changing date format

schemar commented:
Thank you for your contribution @KjellConnelly !

Would you mind splitting out the “mobile toolbar icons” and the “ignore case for queries” changes into their own PRs? That would help a lot and we could merge the simple changes quicker.

Please delete `package-lock.json` and make sure to only use yarn, not npm.

Also, please be patient with the PRs. My time is very limited, but I won’t forget about them!

KjellConnelly commented:
Hi Martin!

Ok here's what I've done: 
- [x] Deleted `package-lock.json` and switch to using yarn
- [x] added `package-lock.json` to `.gitignore`
- [x] updated `tests/Task.test.ts` to test for all supported date formats (3 currently). Also remove duplicate code and made it easier to create new tests should you want to. All tests pass with `yarn run test`
- [x] Ran `yarn run lint` and refactored until no warnings.

Unfortunately, I didn't make separate pull requests for those other two features. However, both features are in the first two commits for this pull request, and are extremely quick to review:
- **ignore case for queries**: 1 line in `src/Query.ts`
- **mobile toolbar icons**: 2 lines in `src/Commands/index.ts`

KjellConnelly commented:
Alright, I just added some new info in `README.md` that describes how to use `Daily-notes` templates, and how `Periodic Notes Plugin` or `Calendar Plugin` can both be used to support the {{date+14d:YYYY-MM-DD}} syntax.

schemar commented:
Thank you @KjellConnelly, but since this PR includes a lot of different additions, it now conflicts with #296. I will manually apply some of the changes from this PR and close it to have the discussion on custom date formats in #296.

schemar commented:
I added the changes to commit 35494cde82bb38c902b9eb4d30dee161e18a1ef0 and added you as co-author. Thank you so much for your effort!
